app/scripts/background.js

```diff
@@ -602,11 +602,20 @@ async function initialize() {
 /**
  * Loads the preinstalled snaps from urls and returns them as an array.
  * It fails if any Snap fails to load in the expected time range.
+ * Supports .json.gz files using gzip decompression.
  */
 async function loadPreinstalledSnaps() {
   const fetchWithTimeout = getFetchWithTimeout();
   const promises = PREINSTALLED_SNAPS_URLS.map(async (url) => {
     const response = await fetchWithTimeout(url);
+
+    // If the Snap is compressed, decompress it
+    if (url.pathname && url.pathname.endsWith('.json.gz')) {
```

Comment:
I don't think it is possible to have a `URL` with no pathname, even if you set the pathname to `null`/`undefined`/`""` or even delete the property it will still never be an empty string or `null`/`undefined`.

```suggestion
    if (url.pathname.endsWith('.json.gz')) {
```